shelley/design-spec/addresses.dot

```diff
@@ -0,0 +1,52 @@
+digraph addresses {
+  graph[compound=true];
+  node [shape=box];
+
+  payment_address [label = "Payment Address"];
+  byron_address [label = "Byron Address"];
+  stake_address [label = "Stake Address"];
+
+  payment_credential [label = "Payment Credential"];
+  stake_credential [label = "Stake Credential"];
```

Comment:
Wasn't our terminology for this the "stake address reference"? It's where a payment address is referring to a stake address (by value, by pointer/ref, or none).

Note otherwise that we have two things called a stake credential.

Reply:
In the formal spec and exec spec, we are saying that and address has a payment credential and a stake reference. Credentials are either a keyhash or a scripthash. The stake reference is either a credential, a pointer, or null.

Reply:
Right! I'll make that change.

Reply:
This needs to be reflected into the tutorials, CLI and code base (these are what you are calling base, pointer or enterprise addresses, right?)

Reply:
Yes, we have

- base address = stake address reference by value
- pointer address = stake address reference by pointer
- enterprise address = null stake address reference.